source/modulo_components/modulo_components/component_interface.py

```diff
@@ -67,29 +67,165 @@ def __step(self):
             self._predicate_publishers[predicate_name].publish(msg)
 
     def __generate_predicate_topic(self, predicate_name: str) -> str:
-        return f'/predicates/{self.get_name()}/{predicate_name}'
+        """
+        Generate the predicate topic name from the name of the predicate.
+
+        :param predicate_name: The predicate name
+        :return: The predicate topic as /predicates/component_name/predicate_name
+        """
+        return f"/predicates/{self.get_name()}/{predicate_name}"
+
+    def add_parameter(self, parameter: Union[str, sr.Parameter], description: str, read_only=False) -> None:
+        """
+        Add a parameter. This method stores either the name of the attribute corresponding to the parameter object or
+        a parameter object directly in the local parameter dictionary and declares the equivalent ROS parameter on the
+        ROS interface. If an attribute name is provided, changing the ROS parameter will also update the provided
+        attribute and vice versa. If the provided argument is not an attribute name or a Parameter object, nothing
+        happens.
```

Comment:
Ah, in my first comment I was just adjusting the grammar of this sentence, but actually it's not quite correct. If the argument is not valid, it throws.

Reply:
Ah yeah right but it will actually not throw anymore once i resolve the TODO below. I will revise all that in a next PR